### ISSUE TITLE
[Bug] 自己复现chat-llava复现不出来

### ISSUE THREAD

**Issue by Cverchen** [OpenGVLab/InternVL#529]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.
- [ ] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.

### Describe the bug

请教的问题所在

### Reproduction

我执行这样的如下的sh文件微调internvl-chat-llava，微调完之后结果与官方提供的相差甚远，其中最令我疑惑的点是预训练的projector是hf上的InternVL-Chat-ViT-6B-Vicuna-7B文件里的么，作者好像说是的。但我用llava的665数据微调后mme才997，请问问题出现在哪里呢？
其中sh文件如下：
#!/usr/bin/env bash
# export PATH='/mllm_hdd/chenjiankang/research_dataset/InternVL/internvl_chat_llava'
set -x

OUTPUT_DIR='/mllm_hdd/chenjiankang/research_dataset/InternVL/internvl_chat_llava/checkpoints/internvl-chat-7b'
GPUS=${GPUS:-8}
NNODES=${NNODES:-1}
NODE_RANK=${NODE_RANK:-0}
PORT=${PORT:-29500}

if [ ! -d "$OUTPUT_DIR" ]; then
  mkdir -p "$OUTPUT_DIR"
fi

echo "Begin running..."
torchrun --nnodes=${NNODES} --nproc_per_node=${GPUS} --master_port=${PORT} \
    /mllm_hdd/chenjiankang/research_dataset/InternVL/internvl_chat_llava/llava/train/train_mem.py \
    --deepspeed /mllm_hdd/chenjiankang/research_dataset/InternVL/internvl_chat_llava/scripts/zero3.json \
    --model_name_or_path /mllm_hdd/chenjiankang/research_dataset/InternVL/internvl_chat_llava/pretrained/vicuna-7b-v1.5 \
    --version v1 \
    --data_path  /mllm_hdd/chenjiankang/research_dataset/LLaVA/playground/data/llava_v1_5_mix665k.json\
    --image_folder /mllm_hdd/chenjiankang/research_dataset/LLaVA/playground/data \
    --vision_tower /mllm_hdd/chenjiankang/research_dataset/InternVL/internvl_chat_llava/pretrained/InternViT-6B-224px \
    --pretrain_mm_mlp_adapter  /mllm_hdd/chenjiankang/research_dataset/InternVL/internvl_chat_llava/pretrained/InternVL-Chat-ViT-6B-Vicuna-7B/mm_projector.bin\
    --mm_projector_type mlp2x_gelu \
    --mm_vision_select_layer -4 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --image_aspect_ratio pad \
    --group_by_modality_length True \
    --bf16 True \
    --output_dir ${OUTPUT_DIR} \
    --num_train_epochs 1 \
    --per_device_train_batch_size 16 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 5198 \
    --save_total_limit 3 \
    --learning_rate 2e-5 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 2048 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --lazy_preprocess True \
    --report_to "tensorboard" \
    | tee ${OUTPUT_DIR}/train.log

### Environment

```Shell
非常感谢！
```

**Comment by Cverchen** [comment]
absl-py                       2.1.0
accelerate                    0.21.0
aiofiles                      23.2.1
aiohttp                       3.9.5
aiosignal                     1.3.1
altair                        5.3.0
annotated-types               0.7.0
anyio                         4.4.0
async-timeout                 4.0.3
attrs                         23.2.0
bitsandbytes                  0.43.1
certifi                       2024.6.2
charset-normalizer            3.3.2
click                         8.1.7
contourpy                     1.2.1
cycler                        0.12.1
dapr                          1.13.0
dapr-ext-fastapi              1.12.0
datasets                      2.20.0
deepspeed                     0.12.6
dill                          0.3.8
distro                        1.9.0
dnspython                     2.6.1
docker-pycreds                0.4.0
einops                        0.6.1
einops-exts                   0.0.4
email_validator               2.2.0
et-xmlfile                    1.1.0
exceptiongroup                1.2.1
fastapi                       0.63.0
fastapi-cli                   0.0.4
ffmpy                         0.3.2
filelock                      3.15.4
flash-attn                    2.6.1
fonttools                     4.53.0
frozenlist                    1.4.1
fsspec                        2024.5.0
gitdb                         4.0.11
GitPython                     3.1.43
googleapis-common-protos      1.63.2
gradio                        4.16.0
gradio_client                 0.8.1
grpcio                        1.64.1
grpcio-status                 1.64.1
h11                           0.14.0
hjson                         3.1.0
httpcore                      0.17.3
httptools                     0.6.1
httpx                         0.24.0
huggingface-hub               0.23.4
idna                          3.7
importlib_resources           6.4.0
jieba                         0.42.1
Jinja2                        3.1.4
joblib                        1.4.2
jsonschema                    4.22.0
jsonschema-specifications     2023.12.1
kess                          0.14.0
kiwisolver                    1.4.5
lfs                           0.2
llava                         1.2.2.post1 /mllm_hdd/chenjiankang/research_dataset/LLaVA
Markdown                      3.7
markdown-it-py                3.0.0
markdown2                     2.4.13
MarkupSafe                    2.1.5
matplotlib                    3.9.0
mdurl                         0.1.2
mpmath                        1.3.0
multidict                     6.0.5
multiprocess                  0.70.16
networkx                      3.3
ninja                         1.11.1.1
nltk                          3.8.1
numpy                         1.26.4
nvidia-cublas-cu12            12.1.3.1
nvidia-cuda-cupti-cu12        12.1.105
nvidia-cuda-nvrtc-cu12        12.1.105
nvidia-cuda-runtime-cu12      12.1.105
nvidia-cudnn-cu12             8.9.2.26
nvidia-cufft-cu12             11.0.2.54
nvidia-curand-cu12            10.3.2.106
nvidia-cusolver-cu12          11.4.5.107
nvidia-cusparse-cu12          12.1.0.106
nvidia-nccl-cu12              2.20.5
nvidia-nvjitlink-cu12         12.5.82
nvidia-nvtx-cu12              12.1.105
openai                        1.35.10
openpyxl                      3.1.5
orjson                        3.10.5
packaging                     24.1
pandas                        2.2.2
peft                          0.11.1
pillow                        10.4.0
pip                           24.1.1
platformdirs                  4.2.2
protobuf                      5.27.2
psutil                        6.0.0
py-cpuinfo                    9.0.0
pyarrow                       16.1.0
pyarrow-hotfix                0.6
pydantic                      1.10.17
pydantic_core                 2.20.0
pydub                         0.25.1
Pygments                      2.18.0
pynvml                        11.5.0
pyparsing                     3.1.2
python-dateutil               2.9.0.post0
python-dotenv                 1.0.1
python-multipart              0.0.9
pytz                          2024.1
PyYAML                        6.0.1
referencing                   0.35.1
regex                         2024.5.15
requests                      2.32.3
rich                          13.7.1
rouge-chinese                 1.0.3
rpds-py                       0.18.1
ruamel.yaml                   0.18.6
ruamel.yaml.clib              0.2.8
ruff                          0.5.0
safetensors                   0.4.3
scikit-learn                  1.2.2
scipy                         1.14.0
semantic-version              2.10.0
sentencepiece                 0.1.99
sentry-sdk                    2.7.1
setproctitle                  1.3.3
setuptools                    69.5.1
shellingham                   1.5.4
shortuuid                     1.0.13
six                           1.16.0
smmap                         5.0.1
sniffio                       1.3.1
starlette                     0.13.6
svgwrite                      1.4.3
sympy                         1.12.1
tensorboard                   2.17.1
tensorboard-data-server       0.7.2
threadpoolctl                 3.5.0
tiktoken                      0.7.0
timm                          0.6.13
tokenizers                    0.15.1
tomlkit                       0.12.0
toolz                         0.12.1
torch                         2.3.1
torchvision                   0.16.2
tqdm                          4.66.4
transformers                  4.37.2
transformers-stream-generator 0.0.5
triton                        2.3.1
typer                         0.12.3
typing_extensions             4.12.2
tzdata                        2024.1
ujson                         5.10.0
urllib3                       2.2.2
uvicorn                       0.30.1
uvloop                        0.19.0
wandb                         0.17.3
watchfiles                    0.22.0
wavedrom                      2.0.3.post3
websockets                    11.0.3
Werkzeug                      3.0.3
wheel                         0.43.0
xformers                      0.0.27
xxhash                        3.4.1
yarl                          1.9.4

**Comment by czczup** [comment]
你好，我用这部分的代码重新跑了一次pretrain + finetune，结果很正常。tensorboard的loss曲线贴在这里供你参考：

<img width="1029" alt="image" src="https://github.com/user-attachments/assets/22b510f3-0237-465f-9400-7b81023d4a35">

**Comment by Cverchen** [comment]
可以放出您跑的pretrain以及finetune的checkpoint么？感谢！

**Comment by czczup** [comment]
我没有计划去开源这个权重，文件也挺大的，这个和之前开源的没有区别。

**Comment by czczup** [comment]
如果你还是跑的有问题，建议把tensorboard贴出来看看

**Comment by Cverchen** [comment]
我finetuning的tensorboard的曲线如下：
<img width="1239" alt="image" src="https://github.com/user-attachments/assets/4c2625a9-b369-43da-8393-b7bcf3971952">

**Comment by Cverchen** [comment]
请问你预训练的数据是558k是吧？

**Comment by Cverchen** [comment]
<img width="1379" alt="image" src="https://github.com/user-attachments/assets/d9882e48-9656-487a-b9aa-e493163d9baa">
finetuning的tensorboard loss曲线

**Comment by Cverchen** [comment]
我预训练好像loss将不到2，只能到3，可能是预训练的问题？

**Comment by czczup** [comment]
> 请问你预训练的数据是558k是吧？

是LLaVA的标注数据，558K的pretrain，665K的finetune

**Comment by Cverchen** [comment]
<img width="1133" alt="image" src="https://github.com/user-attachments/assets/5dba0f1d-dbdf-46ce-bbd9-eb86f4c4d932">
我的pretrain的loss是如下图所示的，数据也是用的558k的pretrain，太迷了呀

**Comment by Cverchen** [comment]
重新创建环境，严格按照环境install可以复现出来了，感谢感谢
